### DISCUSSION TITLE
readOnlyRootFilesystem should be considered high and not low severity

### DISCUSSION THREAD

**Discussion by huornlmj** [aquasecurity/trivy#5806] [Bugs]
### Description

A K8s pod spec with readOnlyRootFilesystem not set to "true" should be flagged as high or critical and not low.  As a pen tester I find that usually the easiest containers to perform exploits in or privilege escalation & escape from are those that have not used readOnlyRootFilesystem = true.  It's a remarkable mitigation and overlooked.  And as Trivy's guidance (https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0014/) points to Kubesec's guidance which says:

> "Without a writable filesystem **the attack surface is dramatically reduced**." 

### Desired Behavior

readOnlyRootFilesystem not being used or not set to true should be high or critical, and not low severity.

### Actual Behavior

readOnlyRootFilesystem not being used is being reported as low severity.

### Reproduction Steps

```bash
1. Create a pod sec with readOnlyRootFilesystem not being used or not set to true 
2. Scan it with Trivy config
3. Observe `LOW: Container 'XXXXX' of Pod 'XXXXX' should set 'securityContext.readOnlyRootFilesystem' to true`
```

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
Cannot share development results here.
```

### Operating System

Linux Ubuntu 22.04

### Version

```bash
$ trivy version
Version: 0.48.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-12-13 06:12:49.269493504 +0000 UTC
  NextUpdate: 2023-12-13 12:12:49.269493123 +0000 UTC
  DownloadedAt: 2023-12-13 10:18:48.266484417 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-09-29 00:52:37.313156528 +0000 UTC
  NextUpdate: 2023-10-02 00:52:37.313156128 +0000 UTC
  DownloadedAt: 2023-09-29 15:01:46.188254631 +0000 UTC
Policy Bundle:
  Digest: sha256:8bfc31f3e4301ef758b6793a07e0b12b4306e0b54d03a640efb2ff5e5ef29b9e
  DownloadedAt: 2023-12-19 11:39:56.147708468 +0000 UTC
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by AnaisUrlichs** [comment]
@simar7 thoughts?

**Comment by simar7** [comment]
Thanks for the suggestion. Seems reasonable to me to bump it up to <kbd>High</kbd> at least.  @chen-keinan what do you think?

**Reply by chen-keinan** [reply]
Sure , it sound reasonable

**Reply by simar7** [reply]
Created an issue to track https://github.com/aquasecurity/trivy/issues/5861
